src/components/ChatEntry.js

```diff
@@ -1,22 +1,49 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.js'
+import {useState} from 'react';
 
 const ChatEntry = (props) => {
+  const onChatLikedButton = () => {
+    const updatedChat = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked 
+    };
+    props.updateEntry(updatedChat);
+
+  };
+
+  const messageAlign = props.sender === 'Vladimir' ? 'chat-entry local' : 'chat-entry remote';
+
+  const heart = props.liked ? '❤️' : '🤍';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={messageAlign}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp}></TimeStamp></p>
+        <button onClick={onChatLikedButton} className="like">{heart}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  entries: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      sender: PropTypes.string.isRequired,
+      body: PropTypes.string.isRequired,
+      timeStamp: PropTypes.string.isRequired,
+      liked: PropTypes.bool.isRequired
+    })
+  ).isRequired,
+  updateChatData: PropTypes.func
```

Comment:
Oo, close! ChatEntry should have different propTypes than ChatLog because it's down a component layer, it should look something like:
```
ChatEntry.propTypes = {
      id: PropTypes.number.isRequired,
      sender: PropTypes.string.isRequired,
      body: PropTypes.string.isRequired,
      timeStamp: PropTypes.string.isRequired,
      liked: PropTypes.bool.isRequired
      updateEntry: PropTypes.func
}
```

☝️ This would also remove the pesky `Failed prop type` console messages that show up when you run the tests!